### PR TITLE
Remove pointers for slices preferring omitempty tag instead 

### PR DESCRIPTION
**Pull request by mrunalp**
Converting to pointers to be consistent with other cgroups settings.

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

**Comment by mrunalp** [comment]
@crosbymichael @vishh PTAL

**Comment by mrunalp** [comment]
We can wait for #284 to merge before this.

File: runtime_config_linux.go [hunk 1]
```diff
@@ -195,7 +195,7 @@ type Network struct {
 	// Set class identifier for container's network packets
 	ClassID *uint32 `json:"classID"`
 	// Set priority of network traffic for container
-	Priorities []InterfacePriority `json:"priorities"`
+	Priorities []*InterfacePriority `json:"priorities"`
```
Comment (vishh) [inline runtime_config_linux.go]: Why? If we add an `omitempty` tag, even an empty list signifies non-existence of this field.

File: runtime_config_linux.go [hunk 2]
```diff
@@ -213,7 +213,7 @@ type Resources struct {
 	// BlockIO restriction configuration
 	BlockIO *BlockIO `json:"blockIO,omitempty"`
 	// Hugetlb limit (in bytes)
-	HugepageLimits []HugepageLimit `json:"hugepageLimits,omitempty"`
+	HugepageLimits []*HugepageLimit `json:"hugepageLimits,omitempty"`
```
Comment (vishh) [inline runtime_config_linux.go]: ditto

File: runtime_config_linux.go (hunk 1 above)
Comment (mrunalp) [inline runtime_config_linux.go]: @vishh I am fine one way or the other. But we have them as pointers in more places compared to where we don't.

Comment (vishh) [inline runtime_config_linux.go]: In general, we don't need pointers for slices and maps. I can go cleanup
the Spec again once I get some spare cycles.

File: runtime_config_linux.go [hunk 3]
```diff
@@ -195,7 +195,7 @@ type Network struct {
 	// Set class identifier for container's network packets
 	ClassID *uint32 `json:"classID"`
 	// Set priority of network traffic for container
-	Priorities []InterfacePriority `json:"priorities"`
+	Priorities []*InterfacePriority `json:"priorities,omitempty"`
```
Comment (vishh) [inline runtime_config_linux.go]: For this PR just adding `omitempty` here should do. Kindly revert the pointer changes.

File: runtime_config_linux.go (hunk 1 above)
Comment (mrunalp) [inline runtime_config_linux.go]: @vishh No issues going one way or the other. I can change the PR.

Comment (vishh) [inline runtime_config_linux.go]: Thanks :)

**Comment by mrunalp** [comment]
@vishh Updated to remove pointers from slices.

**Comment by vishh** [comment]
LGTM. Thanks for the cleanup @mrunalp :+1:

**Comment by hqhq** [comment]
LGTM

**Pull request merged**